libs/api/libbpf_program.cpp

```diff
@@ -390,7 +399,12 @@ bpf_object__pin_programs(struct bpf_object* obj, const char* path)
         char buf[PATH_MAX];
         int len;
 
-        len = snprintf(buf, PATH_MAX, "%s/%s", path, __bpf_program__pin_name(prog));
+        char* pin_name = __bpf_program__pin_name(prog);
```

Comment:
unrelated to changes in this PR: we are allocating memory in failure path. Instead should we instead save the pin paths generated when pinning the programs, and use the same in the failure case?